src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -225,147 +225,4 @@ to_cluster_type(const creatable_topic& t) {
     return ret;
 }
 
-template<typename T>
-static ss::sstring from_config_type(const T& v) {
-    if constexpr (std::is_enum_v<T>) {
-        return ssx::sformat("{}", v);
-    } else if constexpr (std::is_same_v<bool, T>) {
-        return v ? "true" : "false";
-    } else if constexpr (std::is_same_v<T, std::chrono::milliseconds>) {
-        return ss::to_sstring(
-          std::chrono::duration_cast<std::chrono::milliseconds>(v).count());
-    } else {
-        return ss::to_sstring(v);
-    }
-}
-
-config_map_t from_cluster_type(const cluster::topic_properties& properties) {
-    config_map_t config_entries;
-    if (properties.compression) {
-        config_entries[topic_property_compression] = from_config_type(
-          *properties.compression);
-    }
-    if (properties.cleanup_policy_bitflags) {
```

Comment:
Note that by switching to the method in DescribeConfigs, we’re also removing from the response some configs that are not considered in the now-shared code but were considered previously:
 * topic_property_compaction_strategy (compaction.strategy)
 * topic_property_recovery (redpanda.remote.recovery)
 * topic_property_read_replica (redpanda.remote.readreplica)

However, these configs seem to be Redpanda-specific, they are not returned by Apache Kafka. So I think we are not required to return them, although we could choose to return them later as an improvement.

Reply:
Maybe they should have always been included in `DescribeConfigs` as well